hack/test/e2e.sh

```diff
@@ -0,0 +1,251 @@
+
+  sleep 3
+  ./tmp/bin/promq -t http://localhost:30001/metrics  \
+    -q 'controller_runtime_reconcile_errors_total' -o yaml
+
+  local reconcile_errors
+  reconcile_errors=$( ./tmp/bin/promq \
+    -t http://localhost:30001/metrics  -o yaml \
+    -q 'sum(controller_runtime_reconcile_errors_total)' \
+    | tr -d ' ' | grep ^v: | cut -f2  -d: )
+
+  info "Reconcile errors [$stage]: $reconcile_errors"
+
+  kubectl logs -n operators deploy/observability-operator > "$log_file"
+
+  if [[ "$reconcile_errors" -eq 0 ]]; then
+    ok "0 reconciliation errors 🥳"
+    return 0
+  fi
+
+  info "Reconciliation Errors 😱😞"
+  echo ================================================================
+  grep error "$log_file"
+  echo ================================================================
+  warn "Expected 0 reconcile errors but found $reconcile_errors 😱😞"
+  return 1
+
+}
+
+run_bundle(){
+  header "Running ObO Bundle"
+
+ ./tmp/bin/operator-sdk run bundle "$BUNDLE_IMG" \
+   --install-mode AllNamespaces --namespace operators --skip-tls
+}
+
+run_e2e(){
+  header "Running e2e tests"
+  info  " ❯ kc logs -f deploy/observability-operator | ag error"
+
+
+  (kubectl get events -w -n e2e-tests | tee "$LOGS_DIR/events.log")&
+  local events_pid=$!
+
+  go test -v -failfast ./test/e2e/... \
+    --retain=true 2>&1 | tee "$LOGS_DIR/e2e.log" || true
+
+  kubectl logs -n operators deploy/observability-operator |
+    tee "$LOGS_DIR/operator.log" |
+    ag error | tee "$LOGS_DIR/operator-errors.log"
+
+  kill -INT $events_pid
+}
+
+
+declare NO_DEPLOY=false
+
+parse_args() {
+  ### while there are args parse them
+  while [[ -n "${1+xxx}" ]]; do
+    case $1 in
+    -h|--help)      SHOW_USAGE=true; break ;; # exit the loop
+    --no-deploy)    NO_DEPLOY=true; shift ;;
+    *)              return 1 ;; # show usage on everything else
+    esac
+  done
+  return 0
+}
+
+
+print_usage() {
+  local scr
+  scr="$(basename "$0")"
+
+  read -r -d '' help <<-EOF_HELP || true
+Usage:
+  $scr
+  $scr  --no-deploy
+  $scr  -h|--help
+
+
+Options:
+  -h|--help        show this help
+  --no-deploy      do not build and deploy 0b0, useful for rerunning tests
+
+
+EOF_HELP
+
+  echo -e "$help"
+  return 0
+}
+
+
+declare LOGS_DIR="tmp/e2e"
+
+init_logs_dir(){
+  rm -rf "$LOGS_DIR-prev"
+  mv "$LOGS_DIR" "$LOGS_DIR-prev" || true
+  mkdir -p "$LOGS_DIR"
+}
+
+restart_obo() {
+  kubectl scale -n operators --replicas=0 deploy/observability-operator
+  sleep 3
+  kubectl scale -n operators --replicas=1 deploy/observability-operator
+  kubectl wait -n operators --for=condition=Available deploy/observability-operator --timeout=300s
+
+}
+
+create_platform_mon_crds() {
+  kubectl create -k deploy/crds/kubernetes || true
+}
+
+deploy_obo(){
+  delete_olm_subscription || true
+  create_platform_mon_crds
+  build_bundle
+  push_bundle
+  run_bundle
+  expose_metrics
+}
+
+ensure_obo_is_always_pulled() {
+
+  local pull_policy
+  pull_policy=$(kubectl get deploy observability-operator \
```

Comment:
Since the Operator Deployment is done my OLM, it may revert back and changes we make, so it is better to ensure that the source is patched. I thought making that change manually may be help the one running the script to later revert it back. 

Also remember that the script can be run with `--no-deploy` option to test against an already deployed ObO.

Reply:
sorry for my ignorance but isn't there a way to tell OLM to avoid reconciling the deployment (like we have with CVO and overrides)?

Reply:
I am not aware of that as well :(. We should ask the olm team but please note that `imagePullPolicy: Always` only matters if we are rerunning e2e after a new operator image has been pushed to local-registry running in kind.